src/containers/BookEdit.js

```diff
@@ -0,0 +1,102 @@
+import React, { Component, Fragment } from 'react';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import BookShape from '../shapes/book';
+import { fetchBookEdit, updateBook, persistBook } from '../redux/actions';
+
+class BookEdit extends Component {
+  componentDidMount() {
+    this.props.fetchBook(this.props.match.params.isbn);
+  }
+
+  onChangeInput = (event) => {
+    const newBook = {
+      ...this.props.book,
+      [event.target.name]: event.target.value
+    };
+    this.props.updateBook(newBook);
+  }
+
+  onSubmit = (event) => {
+    event.preventDefault();
+    this.props.persistBook(this.props.book)
+      .then(() => this.props.history.push(this.bookDetailPath()));
+  }
+
+  bookDetailPath = () => `/books/${this.props.book.isbn}`
+
+  render() {
+    const { book, loading, error } = this.props;
+    return (
+      <Fragment>
+        { loading ? <div>Loading book...</div> : null }
+        { error ? <div>Could not load book.</div> : null }
+        {
+          book ?
+            <form onSubmit={this.onSubmit}>
```

Comment:
I would also recommend to keep renderable jsx in functional components and only compose them together in a container

Reply:
since class component tends to become a mess, I realize its a demo, but its a very important practice

Reply:
Good point. I added an additional component for the form. Thx!